cleanlab/multilabel_classification/dataset.py

```diff
@@ -0,0 +1,91 @@
+import pandas as pd
+import numpy as np
+from cleanlab.filter import _find_multilabel_issues_per_class
+from cleanlab.internal.multilabel_utils import get_onehot_num_classes
+from collections import defaultdict
+
+
+def common_multilabel_issues(
+    labels=list,
+    pred_probs=None,
+    *,
+    class_names=None,
+    confident_joint=None,
+) -> pd.DataFrame:
+    """Summarizes which classes in a multi-label dataset appear most often mislabeled overall.
+
+    This method works by providing any one (and only one) of the following inputs:
+
+    1. ``labels`` and ``pred_probs``, or
+    2. ``confident_joint``
+
+    Only provide **exactly one of the above input options**, do not provide a combination.
+
+    Parameters
+    ----------
+    labels : List[List[int]]
+        Refer to documentation for this argument in :py:func:`filter._find_multilabel_issues_per_class <cleanlab.filter._find_multilabel_issues_per_class>` for further details.
+
+    pred_probs : np.ndarray, optional
+      Refer to documentation for this argument in :py:func:`filter._find_multilabel_issues_per_class <cleanlab.filter._find_multilabel_issues_per_class>` for further details.
+
+
+    class_names : Iterable[str]
+        A list or other iterable of the string class names. The list should be in the order that
+        matches the label indices. So if class 0 is 'dog' and class 1 is 'cat', then
+        ``class_names = ['dog', 'cat']``.
+
+    confident_joint : np.ndarray, optional
+        Refer to documentation for this argument in :py:func:`filter._find_multilabel_issues_per_class <cleanlab.filter._find_multilabel_issues_per_class>` for further details.
+
+
+    Returns
+    -------
+    common_multilabel_issues : pd.DataFrame
+        DataFrame where each row corresponds to a Class (specified as the row-index) with columns "In Given Label", "In Suggested Label", "Num Examples", "Issue Probability".
+
+        * *In Given Label*: specifies whether the Class is True/False in the given label
+        * *In Suggested Label*: specifies whether the Class is  True/False in the suggested label (based on model prediction)
+        * *Num Examples*: Estimated number of examples with a label issue where this Class is True/False as specified "In Given Label" but cleanlab suggests it should be as specified In Suggested Label. I.e. the number of examples in your dataset where the Class was labeled as True but likely should have been False (or vice versa).
+        * *Issue Probability*: This is the  *Num Examples* column divided by the total number of examples in the dataset. It corresponds to the relative overall frequency of each type of label issue in your dataset.
+
+        By default, the rows in this DataFrame are ordered by "Issue Probability" (descending).
```

Comment:
You didn't describe what happens if the user provides `class_names`

Reply:
https://github.com/cleanlab/cleanlab/blob/3c1524496ec519d772cc322da637b80badfe764a/cleanlab/multilabel_classification/dataset.py#L52